### PR TITLE
Timeline interaction improvement

### PR DESCRIPTION
**Pull request by Eism**
Resolves: #7659

### CODE REVIEW

File: src/playback/internal/au3/au3player.cpp [hunk 1]
```diff
@@ -220,7 +220,7 @@ muse::Ret Au3Player::doPlayTracks(TrackList& trackList, double startTime, double
     return success ? muse::make_ok() : muse::make_ret(muse::Ret::Code::InternalError);
 }
 
-void Au3Player::seek(const muse::secs_t newPosition)
+void Au3Player::seek(const muse::secs_t newPosition, bool applyIfPlaying)
 {
     LOGD() << "newPosition: " << newPosition;
```
Comment (kryksyh) [inline src/playback/internal/au3/au3player.cpp:226]: Is this comment still relevant?

File: src/playback/internal/au3/au3player.cpp [hunk 2]
```diff
@@ -57,7 +57,8 @@ void Au3Player::play()
 
     Au3Project& project = projectRef();
 
-    auto options = ProjectAudioIO::GetDefaultOptions(project, true /*newDefault*/);
+    const bool newDefault = true; //(mode == PlayMode::loopedPlay);
```
Comment (kryksyh) [inline src/playback/internal/au3/au3player.cpp:60]: I don't really understand the meaning of `newDefault` but can you confirm that setting it to `True` from previous `False` does not break anything down the function?

File: src/playback/internal/playbackcontroller.cpp [hunk 3]
```diff
@@ -230,18 +232,24 @@ void PlaybackController::onSeekAction(const muse::actions::ActionData& args)
...
 
-    if (isPlaying()) {
-        LOGD() << "Can't do seek while playing";
-        return;
-    }
-
     muse::secs_t secs = args.arg<double>(0);
-    m_lastPlaybackSeekTime = secs;
-    player()->seek(secs);
+    bool triggerPlay = args.count() > 1 ? args.arg<bool>(1) : false;
```
Comment (kryksyh) [inline src/playback/internal/playbackcontroller.cpp:236]: I can't find where the action is dispatched with a boolean argument, am i missing anything?

File: src/playback/internal/au3/au3player.cpp (hunk 2 above)
Comment (Eism) [reply src/playback/internal/au3/au3player.cpp:60]: actually, newDefault should have always been true if no loop play, like in au3

File: src/playback/internal/playbackcontroller.cpp [hunk 4]
```diff
@@ -48,21 +41,25 @@ void PlaybackController::init()
...
         m_isPlayingChanged.notify();
     });
 
+    m_player->playbackPositionChanged().onReceive(this, [this](const muse::secs_t& position) {
+        project::IAudacityProjectPtr project = globalContext()->currentProject();
+        if (!project) {
+            return false;
+        }
+
+        if (totalPlayTime() == position) {
```
[the remark this answers is not in the document]
Comment (Eism) [reply src/playback/internal/playbackcontroller.cpp:57]: it is two secs_t with muse::is_equal in compare operation

File: src/playback/internal/au3/au3player.cpp (hunk 1 above)
Comment (Eism) [reply src/playback/internal/au3/au3player.cpp:226]: yes, useful

File: src/projectscene/view/playcursor/playcursorcontroller.cpp [hunk 5]
```diff
@@ -39,16 +39,22 @@ void PlayCursorController::init()
...
+        return;
+    }
+
     IProjectViewStatePtr viewState = projectViewState();
     bool snapEnabled = viewState ? viewState->isSnapEnabled() : false;
 
     double secs = m_context->positionToTime(x, snapEnabled);
     if (muse::RealIsEqualOrMore(secs, 0.0)) {
-        dispatcher()->dispatch("playback_seek", ActionData::make_arg1<double>(secs));
+        dispatcher()->dispatch("playback_seek", ActionData::make_arg2<double, bool>(secs, triggerPlay));
```
[the remark this answers is not in the document]
Comment (Eism) [reply src/projectscene/view/playcursor/playcursorcontroller.cpp]: renamed

File: src/playback/internal/au3/au3player.cpp (hunk 2 above)
Comment (kryksyh) [reply src/playback/internal/au3/au3player.cpp:60]: why do we need `if (newDefault)` on line 114 then if it is always true?

File: src/playback/internal/au3/au3player.cpp (hunk 1 above)
Comment (kryksyh) [reply src/playback/internal/au3/au3player.cpp:226]: My bad, the comment in question is not shown on the PR page. I was talking about 

    //! TODO At the moment not work
    //! there probably should be a different implementation
    
    It does work now, right?

Comment (Eism) [reply src/playback/internal/au3/au3player.cpp:226]: ah, yes, no relevant, I think

File: src/playback/internal/au3/au3player.cpp (hunk 2 above)
Comment (kryksyh) [reply src/playback/internal/au3/au3player.cpp:60]: Okay got it, I did not dive deep into what the code is doing, and the name of the varialbe just resets my brain when I'm trying to understand it. 
So the idea is to add more logic to it later, and keep it as it is for now?

**Review by kryksyh — APPROVED** [review]

File: src/playback/internal/au3/au3player.cpp (hunk 2 above)
Comment (Eism) [reply src/playback/internal/au3/au3player.cpp:60]: yes

**Comment by chinakov** [comment]
Tested:
- Shortcuts for playback (Spacebar, Shift+Spacebar) 
- Bottom section of the Timeline allows for playback

Omitted from testing: Looping selection cases (Ctrl+Spacebar, Top Timeline Bar)

**Pull request merged**